### PR TITLE
feat(docs,core): Adds error reference pages and populate codes on errors in core

### PR DESCRIPTION
**Pull request by jacoblee93**
@efriis

### CODE REVIEW

File: docs/core_docs/docs/troubleshooting/errors/GRAPH_RECURSION_LIMIT.mdx [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# GRAPH_RECURSION_LIMIT
+
+Your LangGraph app reached the maximum number of steps before hitting a stop condition.
+
+This is usually a sign of an infinite loop, but complex graphs may hit this naturally.
```
Comment (efriis) [inline docs/core_docs/docs/troubleshooting/errors/GRAPH_RECURSION_LIMIT.mdx:5]: should we have an example repro of how to trigger this error for mental pattern matching?

File: docs/core_docs/docs/troubleshooting/errors/INVALID_GRAPH_UPDATE.mdx [hunk 2]
```diff
@@ -0,0 +1,40 @@
+# INVALID_GRAPH_UPDATE
+
+A LangGraph app received an unexpected return value from a node and failed to update its state at the end of a step.
```
Comment (efriis) [inline docs/core_docs/docs/troubleshooting/errors/INVALID_GRAPH_UPDATE.mdx:3]: are we certain this is the only condition that creates this? can it also be an input with a thread?

Comment (jacoblee93) [reply docs/core_docs/docs/troubleshooting/errors/INVALID_GRAPH_UPDATE.mdx:3]: Hmm probably not, might be worth splitting the error more granularly

Comment (jacoblee93) [reply docs/core_docs/docs/troubleshooting/errors/INVALID_GRAPH_UPDATE.mdx:3]: Yeah let's actually split this into different cases and keep this focused on parallel updates

**Pull request merged**